Compilation failure with `aeson-2`

To stay in Stackage, we need to upgrade to `aeson-2`.  
Currently, this produces a build failure, so we cannot simply relax the upper bound.
```
[332 of 404] Compiling Agda.Interaction.JSON ( src/full/Agda/Interaction/JSON.hs, /Users/abel/agda-erasure/dist-newstyle/build/x86_64-osx/ghc-9.0.1/Agda-2.6.3/build/Agda/Interaction/JSON.o, /Users/abel/agda-erasure/dist-newstyle/build/x86_64-osx/ghc-9.0.1/Agda-2.6.3/build/Agda/Interaction/JSON.dyn_o ) [Data.Aeson.Types changed]

src/full/Agda/Interaction/JSON.hs:47:10: error:
    • Couldn't match expected type ‘Key’ with actual type ‘Text’
    • In the first argument of ‘(.=)’, namely ‘key’
      In the second argument of ‘($)’, namely ‘key .= toJSON value’
      In a stmt of a 'do' block: pure $ key .= toJSON value
   |
47 |   pure $ key .= toJSON value
   |          ^^^
```

--- a/Agda.cabal
+++ b/Agda.cabal
@@ -203,7 +203,7 @@ library
   if os(windows) && impl(ghc == 8.6.3)
     buildable: False
 
-  build-depends:  aeson >= 1.1.2.0 && < 1.6
+  build-depends:  aeson >= 1.1.2.0 && < 2.1
                 , array >= 0.5.1.1 && < 0.6
                 , async >= 2.2 && < 2.3
                 , base >= 4.9.0.0 && < 4.16

--- a/src/full/Agda/Interaction/Highlighting/JSON.hs
+++ b/src/full/Agda/Interaction/Highlighting/JSON.hs
@@ -12,7 +12,6 @@ import Agda.TypeChecking.Monad (HighlightingMethod(..), ModuleToSource)
 import Agda.Utils.FileName (filePath)
 import Agda.Utils.IO.TempFile (writeToTempFile)
 
-import Data.Aeson
 import qualified Data.ByteString.Lazy.Char8 as BS
 import qualified Data.Map as Map
 

--- a/src/full/Agda/Interaction/JSON.hs
+++ b/src/full/Agda/Interaction/JSON.hs
@@ -4,16 +4,22 @@
 -- | Encoding stuff into JSON values in TCM
 
 module Agda.Interaction.JSON
-  ( EncodeTCM(..)
+  ( module Export
+  , EncodeTCM(..)
   , obj, kind, kind'
+  , (.=)
   , (@=), (#=)
-  , (>=>), (<=<)
-  -- , ToRep(..), rep
   ) where
 
-import Control.Monad ((>=>), (<=<))
-import Data.Aeson
-import Data.Aeson.Types (Pair)
+import Control.Monad as Export ((>=>), (<=<))
+import Data.Aeson    as Export hiding (Result(..), (.=))
+
+import qualified Data.Aeson
+import Data.Aeson.Types ( Pair )
+#if MIN_VERSION_aeson(2,0,0)
+import qualified Data.Aeson.Key as Key
+#endif
+
 import Data.Text (Text)
 import GHC.Int (Int32)
 
@@ -27,6 +33,16 @@ import Agda.Utils.Pretty
 import qualified Agda.Utils.FileName as File
 import qualified Agda.Utils.Maybe.Strict as Strict
 
+#if MIN_VERSION_aeson(2,0,0)
+toKey :: Text -> Key
+toKey = Key.fromText
+#else
+type Key = Text
+
+toKey :: Text -> Key
+toKey = id
+#endif
+
 ---------------------------------------------------------------------------
 -- * The EncodeTCM class
 
@@ -39,6 +55,10 @@ class EncodeTCM a where
 -- | TCM monadic version of object
 obj :: [TCM Pair] -> TCM Value
 obj = (object <$>) . sequence
+
+-- | A key-value pair for encoding a JSON object.
+(.=) :: ToJSON a => Text -> a -> Pair
+(.=) = (Data.Aeson..=) . toKey
 
 -- | Pairs a key with a value wrapped in TCM
 (#=) :: (ToJSON a) => Text -> TCM a -> TCM Pair

--- a/src/full/Agda/Interaction/JSONTop.hs
+++ b/src/full/Agda/Interaction/JSONTop.hs
@@ -3,7 +3,6 @@ module Agda.Interaction.JSONTop
     ) where
 import Control.Monad.State
 
-import Data.Aeson hiding (Result(..))
 import Data.ByteString.Lazy (ByteString)
 import qualified Data.ByteString.Lazy.Char8 as BS
 import qualified Data.Text as T

